script/tool/test/update_excerpts_command_test.dart

```diff
@@ -447,6 +443,74 @@ FAIL
       ]),
     );
   });
+
+  group('File type tests', () {
+    const List<Map<String, String>> testCases = <Map<String, String>>[
+      <String, String>{'fileName': 'main.cc'},
+      <String, String>{'fileName': 'main.cpp', 'language': 'c++'},
+      <String, String>{'fileName': 'main.dart'},
+      <String, String>{'fileName': 'main.js'},
+      <String, String>{'fileName': 'main.kt', 'language': 'kotlin'},
+      <String, String>{'fileName': 'main.java'},
+      <String, String>{'fileName': 'main.gradle', 'language': 'groovy'},
+      <String, String>{'fileName': 'main.m', 'language': 'objectivec'},
+      <String, String>{'fileName': 'main.swift'},
+      <String, String>{
+        'fileName': 'main.css',
+        'prefix': '/* ',
+        'suffix': ' */'
+      },
+      <String, String>{
+        'fileName': 'main.html',
+        'prefix': '<!--',
+        'suffix': '-->'
+      },
+      <String, String>{
+        'fileName': 'main.xml',
+        'prefix': '<!--',
+        'suffix': '-->'
+      },
+      <String, String>{'fileName': 'main.yaml', 'prefix': '# '},
+      <String, String>{'fileName': 'main.sh', 'prefix': '# '},
+      <String, String>{'fileName': 'main', 'language': 'txt', 'prefix': ''},
+    ];
+
+    void runTest(Map<String, String> testCase) {
+      test('updates ${testCase['fileName']} files', () async {
+        final String fileName = testCase['fileName']!;
+        final String language = testCase['language'] ?? fileName.split('.')[1];
+        final String prefix = testCase['prefix'] ?? '// ';
+        final String suffix = testCase['suffix'] ?? '';
+
+        await testInjection('''
+Example:
+
+<?code-excerpt "$fileName (SomeSection)"?>
+```$language
+X Y Z
+```
+''', '''
+FAIL
+$prefix#docregion SomeSection$suffix
+A B C
+$prefix#enddocregion SomeSection$suffix
+FAIL
+''', '''
+Example:
+
+<?code-excerpt "$fileName (SomeSection)"?>
+```$language
+A B C
+```
+''', fileName);
+      });
+    }
+
+    // ignore: prefer_foreach
```

Comment:
Two conflicting lint rules here: [avoid_function_literals_in_foreach_calls](https://dart.dev/tools/linter-rules/avoid_function_literals_in_foreach_calls) and [prefer_foreach](https://dart.dev/tools/linter-rules/prefer_foreach). The latter aims to make the code more terse but the formatter uses the same number of lines with `forEach` as it does with `for...in`.

Reply:
There's not a conflict, the linter just gives a misleading suggestion for what you probably wrote as your other version. The version of this that's compliant with all of our lints is `testCases.forEach(runTest);`.